src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -113,7 +119,10 @@ public static TestInputConfiguration parseWithFilteredViolations(String inputFil
 
     private static void setModules(TestInputConfiguration.Builder testInputConfigBuilder,
                                    String inputFilePath, List<String> lines)
-            throws Exception {
+            throws CheckstyleException {
+        if (!lines.get(0).startsWith("/*")) {
+            throw new CheckstyleException("Config not specified on top.");
```

Comment:
> BDD not in correct position

We cannot determine that.
We simply start checking from the top and assume everything is written ideally, if it isn't we throw an exception.

> Checkstyle requires a BDD configuration at the first line in the input file detailing the configuration file to use in processing the file for violations. This was not found or found in the correct position and needs to be resolved for the file to processed correctly.

Sounds good to me.

> If @romani agrees, we should document the specifics somewhere.

I discussed this already with @romani . We seemed to agree that future developers shall follow by example. However, I'll do it if we decide in favour.

Reply:
I agree with with following by example, but I also see some weird cases where it may not be clear where to find an example. Example of me asking if we support 2+ modules in BDD for one config.

Again documentation can be another issue if @romani agrees, and in the meantime we can use some wording like `Please see other inputs for examples of what is required.`

Reply:
Okay, I'll do it after the issue gets approved.

Reply:
issue is approved.

Reply:
I do not belive in documentation when it is possible to show all by numerous examples in code.
Ones we finish all support, @shashwatj07  can write some article to summarize his experience and what he archived.
meanwhile I would rather focus on completion of what we planned to finish.

Reply:
we can place some basic information(our format is very very simple) and extend it later on.
It is better to finish migration, as workiing code is better example than tons of documentations.

Reply:
>  and in the meantime we can use some wording like Please see other inputs for examples of what is required.

done